### PR TITLE
🚀 Layers: Lazily remeasure child layouts

### PR DESCRIPTION
**Pull request by jridgewell**
This is attempt 2 to fix the quadratic remeasurements described in #22809.

This changes the layers code so that child layouts are remeasured as needed, not eagerly when the parent layer is remeasured. Still, causing any mutation inside a layer causes all layouts inside that layer to be invalidated. But since we no longer need to remeasure everything in that layer to get the measurements of a single layout, this leads to significantly less remeasures overall.

### CODE REVIEW

File: src/service/layers-impl.js [hunk 1]
```diff
@@ -751,6 +751,9 @@ export class LayoutElement {
     // This might lead to a double tracking.
     if (!this.children_.includes(child)) {
       this.children_.push(child);
+      if (child.isLayer()) {
+        this.transfer_(child);
+      }
```
Comment (choumx) [inline src/service/layers-impl.js]: Is this an unrelated bug fix?

Comment (jridgewell) [reply src/service/layers-impl.js]: It's an optimization that allows the parent layer to be discovered during `remeasure` calls (which are obviously measure phases) instead of when declaring an layout is a layer (which may be in a mutate phase).

Discovering the parent layer requires a measure phase because we have to check if an ancestor (or the element itself) has `position: fixed` style.
